Engine/ResultProxy.cs

```diff
@@ -642,7 +650,14 @@ public void Invoke(IResultListener a, WorkQueue queue)
                 }
             }
 
-
+            // Skip the invocation if the result listener does not implement OnResultPublished.
+            public bool Skip(IResultListener a, WorkQueue b)
+            {
+                if (a is ResultListener r)
+                    return ResultListener.ImplementsOnResultsPublished(r) == false;
+                
+                return false;
+            }
```

Comment:
In some scenarios this would have a large effect and in some it would not. So I could make a benchmark that shows a large improvement, but I don't know how relevant that would be.